src/python/pants/engine/internals/scheduler.py

```diff
@@ -621,6 +621,19 @@ def digests_to_bytes(self, digests: Sequence[Digest]) -> Tuple[bytes]:
             tuple(self._scheduler._native.lib.digests_to_bytes(sched_pointer, list(digests))),
         )
 
+    def snapshots_to_file_contents(
+        self, snapshots: Sequence[Snapshot]
+    ) -> Tuple[Tuple[FileContent]]:
+        sched_pointer = self._scheduler._scheduler
+        return cast(
+            Tuple[Tuple[FileContent]],
+            tuple(
+                self._scheduler._native.lib.snapshots_to_file_contents(
+                    sched_pointer, list(snapshots)
+                )
+            ),
+        )
```

Comment:
I don't think we need a special Rust entry point. I think you can use the synchronous API to use the equivalent of:

```python
result = await MultiGet(Get(DigestContents, Digest, snapshot.digest) for snapshot in snapshots)
```

It doesn't look like the Rust code is doing more than converting a `Snapshot` -> sequence of `FileContent`, right?